### PR TITLE
Add compatibility with Gradle 6

### PR DESCRIPTION
**Pull request by chali**

### CODE REVIEW

File: src/main/groovy/nebula/plugin/publishing/ivy/PlatformDependencyVerifier.groovy [hunk 1]
```diff
@@ -52,7 +52,6 @@ class PlatformDependencyVerifier {
         Map<String, Set<? extends ModuleIdentifier>> dependencyMap = [:]
         dependencyMap['runtime'] = platformDependencies(configurations.runtimeClasspath)
         dependencyMap['compile'] = platformDependencies(configurations.compileClasspath)
-        dependencyMap['compileOnly'] = platformDependencies(configurations.compileOnly)
```
Comment (chali) [inline src/main/groovy/nebula/plugin/publishing/ivy/PlatformDependencyVerifier.groovy]: We don't see people putting recommendation into `compileOnly` right?

File: src/test/groovy/nebula/plugin/publishing/maven/MavenBasePublishPluginIntegrationSpec.groovy [hunk 2]
```diff
@@ -164,38 +164,30 @@ class MavenBasePublishPluginIntegrationSpec extends IntegrationTestKitSpec {
...
-              <name>maventest</name>
-            </project>
-        '''.stripIndent()
-
         when:
         runTasks('publishNebulaPublicationToTestLocalRepository')
 
         then:
-        new File(publishDir, 'maventest-0.1.0.pom').text == expectedPom
+        def resultPom = new File(publishDir, 'maventest-0.1.0.pom').text
```
Comment (chali) [inline src/test/groovy/nebula/plugin/publishing/maven/MavenBasePublishPluginIntegrationSpec.groovy]: Gradle 6 adds marker info so I'm trying to find common ground between Gradle 5 and 6

File: src/main/groovy/nebula/plugin/publishing/ivy/PlatformDependencyVerifier.groovy (hunk 1 above)
Comment (rpalcolea) [reply src/main/groovy/nebula/plugin/publishing/ivy/PlatformDependencyVerifier.groovy]: not really, it is weird.

File: src/test/groovy/nebula/plugin/publishing/maven/MavenBasePublishPluginIntegrationSpec.groovy (hunk 2 above)
Comment (rpalcolea) [reply src/test/groovy/nebula/plugin/publishing/maven/MavenBasePublishPluginIntegrationSpec.groovy]: yeah, string comparison won't work unless gradle metadata is disabled but this is safer.

Comment (OdysseusLives) [reply src/test/groovy/nebula/plugin/publishing/maven/MavenBasePublishPluginIntegrationSpec.groovy]: +1 makes sense

**Pull request merged**